pkg/templates/pdbminavailable/template.go

```diff
@@ -0,0 +1,173 @@
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/extract"
+	"golang.stackrox.io/kube-linter/pkg/k8sutil"
+	"golang.stackrox.io/kube-linter/pkg/lintcontext"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/pdbminavailable/internal/params"
+	pdbV1 "k8s.io/api/policy/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	templateKey = "pdb-min-available"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Pod Disruption Budget Configuration Checks",
+		Key:         templateKey,
+		Description: "Flag non-optimal PDB configurations",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{
+				objectkinds.PodDisruptionBudget},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return minAvailableCheck, nil
+		}),
+	})
+}
+
+func minAvailableCheck(lintCtx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+
+	var results []diagnostic.Diagnostic
+
+	// Get the PDB provided
+	pdb, ok := object.K8sObject.(*pdbV1.PodDisruptionBudget)
+	if !ok {
+		return nil
+	}
+
+	// If MinAvailable isn't set, then no need to check
+	if pdb.Spec.MinAvailable == nil {
+		return results
+	}
+
+	// Extract the MinAvailable value from the spec and check if it's a number or percentage
+	value, isPercent, err := getIntOrPercentValueSafelyFromString(pdb.Spec.MinAvailable.String())
+	if err != nil {
+		return []diagnostic.Diagnostic{
+			{
+				Message: fmt.Sprintf("PDB has invalid MinAvailable value: %v", err),
+			},
+		}
+	}
+
+	// If the value is a percentage, handle the case where the MinValue is set to 100%
+	// as DeploymentLike's replica counts don't need to be compared
+	if isPercent && value == 100 {
+		return []diagnostic.Diagnostic{
+			{
+				Message: "PDB has minimum available replicas set to 100 percent of replicas",
+			},
+		}
+	}
+
+	// Build the label selector for the PDB to use for comparison
+	labelSelector, err := metaV1.LabelSelectorAsSelector(&metaV1.LabelSelector{
+		MatchLabels:      pdb.Spec.Selector.MatchLabels,
+		MatchExpressions: pdb.Spec.Selector.MatchExpressions},
+	)
+
+	if err != nil {
+		return []diagnostic.Diagnostic{
+			{
+				Message: fmt.Sprintf("PDB has invalid label selector: %v", err),
+			},
+		}
+	}
+
+	// Evaluate Deploymet Likes in the lintContext to see if they have MinAvailable set too low
+	deploymentLikes, err := getDeploymentLikeObjects(lintCtx, labelSelector, pdb.Namespace)
+	if err != nil {
+		return []diagnostic.Diagnostic{
+			{
+				Message: fmt.Sprintf("Failed to retrieve deployments matching the PDB's label selector within namespace %s: %v", pdb.Namespace, err.Error()),
```

Comment:
as above
```suggestion
				Message: fmt.Sprintf("Failed to retrieve deployments matching the PDB's label selector within namespace %s: %w", pdb.Namespace, err.Error()),
```

Reply:
`fmt.Sprintf does not support error-wrapping directive %w`